trio/_core/_run.py

```diff
@@ -1167,24 +1180,27 @@ class Task(metaclass=NoPublicConstructor):
     # Tasks start out unscheduled.
     _next_send_fn = attr.ib(default=None)
     _next_send = attr.ib(default=None)
-    _abort_func = attr.ib(default=None)
-    custom_sleep_data = attr.ib(default=None)
+    _abort_func: Callable[[Callable[[], NoReturn]], Abort] | None = attr.ib(
+        default=None
+    )
+    # could possible be set with a TypeVar
+    custom_sleep_data: Any = attr.ib(default=None)
```

Comment:
see comment

Reply:
Be better to make this `object`, forcing users to do an `isinstance` check each time. Since anything touching the task could have set this, it's not really going to be safe to assume that this had the value you saw it with last.

Reply:
Sounds reasonable, added a comment copying most of your explanation.

Reply:
Hm, getting a bunch of errors in `trio._channel`:
```
trio/_channel.py:181: error: "object" has no attribute "_tasks"  [attr-defined]
trio/_channel.py:281: error: "object" has no attribute "_tasks"  [attr-defined]
trio/_channel.py:318: error: "object" has no attribute "_tasks"  [attr-defined]
trio/_channel.py:427: error: "object" has no attribute "_tasks"  [attr-defined]
```

so internal trio functions are certainly making assumptions about the type.

Reply:
`type: ignore` them, or add `isinstance` checks? (Or put it back to `Any` and leave properly typing it with a `TypeVar` or something for the future)

Reply:
Fwiw in my opinion this and the rescheduling thing are probably the only places I actually think exposing `Any` is a good thing :P 

There's not much you can do if you suddenly find the data has changed, after all. But idk up to you ig

Reply:
I'll set it to `Any` for now, and let future PR make it stricter if anybody cares.